### ISSUE TITLE
Block with long toolbar goes behind block settings

### ISSUE THREAD

**Issue by dsas** [WordPress/gutenberg#39441]
### Description

When a block has `ToolbarButton` that are too wide to fit on the viewport, it should scroll horizontally until the 'three dots' block settings menu is displayed at the far right of the Toolbar. If there is an intermediary element, e.g. a div or a `ToolbarGroup` the 'three dots' block settings menu is displayed over it instead.

### Step-by-step reproduction instructions

1. Create a block with an edit property that has several controls inside a ToolbarGroup, such that it's wide.
2. Change the viewport to be narrower than the toolbar created e.g. < 400px wide
3. Go to the new post screen
4. Insert the block

### Screenshots, screen recording, code snippet

With ToolbarGroup
![image](https://user-images.githubusercontent.com/93301/158272774-a9c21fbe-75f3-4b69-a74d-ecacdb31ec23.png)

The expected behaviour - made by removing ToolbarGroup from the menu
https://user-images.githubusercontent.com/93301/158274170-ba939f80-fe1f-4c07-80e6-004ca787c094.mp4

Here's the example menu I used, just paste it over an existing blocks edit method to be able to reproduce quickly
```javascript
edit: () => {
	return (
		<>
			<BlockControls>
				<ToolbarGroup>
					<ToolbarButton>
						Button one
					</ToolbarButton>
					<ToolbarButton>
						Button two
					</ToolbarButton>
					<ToolbarButton>
						Button three
					</ToolbarButton>
					<ToolbarButton>
						Button four
					</ToolbarButton>
					<ToolbarButton>
						Button five
					</ToolbarButton>
				</ToolbarGroup>
			</BlockControls>
		</>
	);
},
```

### Environment info

 - WP 5.9.2
 - Gutenberg trunk - 0907f097 is HEAD

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
I tried this in the table block, as that has quite a small toolbar, and it does seem to be a problem:
![Screen Shot 2022-03-15 at 11 46 15 am](https://user-images.githubusercontent.com/677833/158302212-986a3f9b-6b67-4181-a969-7eec2a5e1c25.png)

The RichText formats are overlapping there, but the same kind of issue.

For some reason it doesn't seem to happen for most core blocks, but that may be because many of them are using an existing Group like this:
```js
<BlockControls group="block">
```

The Navigation block is one that I could find that suffers very slightly from this bug.

**Comment by dsas** [comment]
Yeah, sorry I should have said. Using `<BlockControls group=` is how I fixed the actual issue I was having (https://github.com/Automattic/jetpack/issues/22745)

**Comment by dsas** [comment]
The core block I used for testing was `cover` if that helps.
